In [ ]:
Roll no:108806
Name: Rohith R
course name:DA
batch:geetika mam batch (9:30 to 11:30)

## chi_square test

### 1. state the hypothesis

<p style="color:blue;font-size:15px;font-family:'Times new roman';">a hypothesis test designed to test for a statistically significant relationship between nominal and ordinal variables organized in a bivariate table
it will checks the dependency between the categories.</p>

**Null hypothesis (Ho):**

---> The variables are independent

---> The proportions of one variable are the same for different values of the other variable

---> The population follows the specified distribution


**Alternative hypothesis (H1):**

---> The variables are related

---> The proportions of one variable are not the same for different values of the other variable

---> The population does not follow the specified distribution



### 2. Compute the Chi-Square Statistic 

chi_square_statistic = np.sum((observed - expected)**2 / expected)

- **observed**: An array of observed frequencies for each category.

- **expected:** An array of expected frequencies for each category.

- The np.sum function sums up the individual values across the array.

- Divide the squared difference by the expected frequency for each category.
Sum the results for all categories.

In [12]:
# 3. create the numpy array and pandas dataframe to find the critical value
import numpy as np
observed=np.array([[50,70],[80, 100],
    [60, 90],
    [30, 50],
    [20, 50]
])
observed

array([[ 50,  70],
       [ 80, 100],
       [ 60,  90],
       [ 30,  50],
       [ 20,  50]])

In [31]:
# create the data frame
import pandas as pd
data_table=pd.DataFrame({'satisfaction':['Very Satisfied','Satisfied','Neutral','Unsatisfied','Very Unsatisfied'],
               'Smart Thermostat':[50,80,60,30,20],
             'Smart Light':[70,100,90,50,50]})
data_table

satisfaction  Smart Thermostat  Smart Light
0    Very Satisfied                50           70
1         Satisfied                80          100
2           Neutral                60           90
3       Unsatisfied                30           50
4  Very Unsatisfied                20           50

In [63]:
# observed values
row=observed.sum(axis=1)
row

array([120, 180, 150,  80,  70])

In [65]:
column=observed.sum(axis=0)
column

array([240, 360])

In [67]:
Total=observed.sum()
print(f'The total observed :{Total}')

The total observed :600


In [53]:
# expected values
expected_values=np.outer(row,column)/Total
expected_values

array([[ 48.,  72.],
       [ 72., 108.],
       [ 60.,  90.],
       [ 32.,  48.],
       [ 28.,  42.]])

### 3. Determine the critical value

- **Find Critical Value:** The critical value for the chi-square statistic at alpha=0.05 and df = 4 can be found using a chi-square distribution

- **Find Degrees of Freedom (df):** Using the formula

**df = (number_rows-1)(number_columns-1), given rows are 5 and columns is 2**

- If your calculated chi-square statistic exceeds this critical value, you can reject the null hypothesis. Otherwise, you fail to reject the null hypothesis.

In [78]:
import scipy.stats as stats # import scipy libraries
critical_value=stats.chi2.ppf(q=0.95,df=4) #  1-0.05=0.95 ,    df=(rows-1)(columns-1) ...(5-1)(2-1)=4
print(f'The critical value :{np.round(critical_value,3)}')

The critical value :9.488


### 4. Make a Decision 

- If the Chi-Square statistic is less than critical value, we fail to reject the null hypothesis.

- **5.638<9.488**,we fail to reject the null hypothesis

- The Chi-Square statistic **(5.545)** is less than the critical value **(9.488)** at the **0.050** significance level.

- **we fail to reject the null hypothesis.** This means that there is not enough evidence to suggest that the observed frequencies differ significantly from the expected frequencies.

## Hypothesis Testing

### 1. state the hypothesis

<p style="color:blue;font-size:15px;font-family:'Times new roman';">a hypothesis test designed to test for a statistically significant relationship between nominal and ordinal variables organized in a bivariate table it will checks the dependency between the categories.</p>

**Null hypothesis (Ho):**

---> The variables are independent

---> The proportions of one variable are the same for different values of the other variable

---> The population follows the specified distribution

**Alternative hypothesis (H1):**

---> The variables are related

---> The proportions of one variable are not the same for different values of the other variable

---> The population does not follow the specified distribution

### 2. Calculate the Test Statistic

In [4]:
x_bar = 3050
mu = 4000
n = 25
sigma = 125

In [6]:
import math    # import math library 
Zclac = (x_bar-mu)/(sigma/math.sqrt(n))     # Test statistic formula
print("test statistic value :",Zclac)

test statistic value : -38.0


### 3.Determine the Critical Value 

In [14]:
from scipy.stats import norm
alph=0.05
z_critical=norm.ppf(1-0.05)
z_cr=np.round(z_critical,3)
print(f'The critical_value : {z_cr}')

The critical_value : 1.645


- the critical Z-value for a one-tailed test at a 5% significance level (α = 0.05) is approximately **1.645.**

### 4. Make a Decision 

- The test statistic we calculated is **-38.**

- The critical value for a one-tailed test at alph = 0.05 critical **z_value = 1.645.**

- Since the test statistic -38 is much less than **1.645 (-38<1.645),** it is far in the left tail of the normal distribution.

- The test statistic (-38) is less than the critical value (1.645), so we fail to reject the null hypothesis.

- **Ho is rejected and H1 is accepted**

### 5. Conclusion 

- There is not enough evidence to support the restaurant owner's claim that the weekly operating costs are higher than what the cost model suggests.

- The null hypothesis stated that the average weekly operating cost is 4,000 Rs, as per the cost model

- The alternative hypothesis suggested that the average cost was greater than 4,000 Rs.

- However, the test statistic (-38) was much smaller than the critical value (1.645), indicating that the sample mean of 3,050 Rs is significantly lower than the theoretical value of 4,000 Rs.

- Therefore, we fail to reject the null hypothesis, and there is insufficient statistical evidence to conclude that the operating costs are higher than the theoretical value of 4,000 Rs. The evidence does not support the restaurant owners' claim of increased operating costs